packages/micropip/src/micropip/_micropip.py

```diff
@@ -86,23 +90,28 @@ def is_compatible(self):
                 return True
         return False
 
-    async def download(self, fetch_kwargs):
+    async def _fetch_bytes(self, fetch_kwargs):
         try:
-            wheel_bytes = await fetch_bytes(self.url, fetch_kwargs)
+            return await fetch_bytes(self.url, fetch_kwargs)
         except OSError as e:
-            if self.url.startswith("https://files.pythonhosted.org/"):
+            if self.parsed_url.hostname == "files.pythonhosted.org":
```

Comment:
Maybe also,

```suggestion
            if self.parsed_url.hostname in ["files.pythonhosted.org", "cdn.jsdelivr.net"]:
```
as we have wheels there.

Reply:
Not strictly related to this PR, but sure it seems like an improvement. Maybe store the known-CORS domains into a global variable. Relatedly, we should ask pypi to add CORs to their 404 page...